contracts/interchain-token-service/src/contract/execute.rs

```diff
@@ -126,3 +132,93 @@ pub fn deregister_its_contract(deps: DepsMut, chain: ChainNameRaw) -> Result<Res
 
     Ok(Response::new().add_event(Event::ItsContractDeregistered { chain }.into()))
 }
+
+pub fn register_gateway_token(
+    deps: DepsMut,
+    denom: String,
+    _chain: ChainNameRaw,
+) -> Result<Response, Error> {
+    let token_id = get_token_id(&deps, &denom)?;
+    state::save_gateway_token_denom(deps.storage, token_id, denom)
+        .change_context(Error::FailedGatewayTokenRegistration)?;
+    Ok(Response::new())
+}
+
+pub fn get_token_id(deps: &DepsMut, denom: &str) -> Result<TokenId, Error> {
+    let config = state::load_config(deps.storage);
+    let gateway: axelarnet_gateway::Client =
+        client::ContractClient::new(deps.querier, &config.axelarnet_gateway).into();
+    let chain_name = gateway
+        .chain_name()
+        .change_context(Error::FailedTokenIdGeneration)?;
+    let chain_name_hash: [u8; 32] = Keccak256::digest(chain_name.to_string().as_bytes()).into();
+
+    let gateway_token_prefix: [u8; 32] =
+        Keccak256::digest("its-interchain-token-id-gateway").into();
```

Comment:
```suggestion
        Keccak256::digest("its-interchain-token-id-gateway").into();
```
Define a constant for this?

Reply:
Sorry, I meant to define a constant for the string. Would be great if Rust allowed defining const trait fns but don't see progress on that [RFC](https://github.com/rust-lang/rfcs/pull/3490).